clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -1630,6 +1635,52 @@ def GetMemberOp : CIR_Op<"get_member"> {
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// VecElemOp
+//===----------------------------------------------------------------------===//
+
+def VecElemOp : CIR_Op<"vec_elem", [Pure,
+  TypesMatchWith<"type of 'result' matches element type of 'vec'",
+                 "vec", "result",
+                 "$_self.cast<VectorType>().getEltType()">]> {
+
+  let summary = "Extract one element from a vector object";
+  let description = [{
+    The `cir.vec_elem` operation extracts one element from a vector object.
+  }];
+
+  let arguments = (ins CIR_VectorType:$vec, CIR_IntType:$index);
+  let results = (outs AnyType:$result);
+
+  let assemblyFormat = [{
+    $vec `[` $index `:` type($index) `]` type($vec) `->` type($result) attr-dict
+  }];
+
+  let hasVerifier = 0;
+}
+
+//===----------------------------------------------------------------------===//
+// VecValue
+//===----------------------------------------------------------------------===//
+
+def VecValueOp : CIR_Op<"vec", [Pure]> {
```

Comment:
I found the name a bit too generic, I'd go for `cir.vec.build` or `cir.vec.insert`

Reply:
I admit that `cir.vec` is too generic.  I prefer `cir.vec.create`.  I would be fine with `cir.vec.build` or `cir.vec.construct`.  I object to `cir.vec.insert`, because I want to use that name for assigning to a vector element, as in `v[1] = s`, where a single scalar value is inserted into an existing vector.